src/views/VideoView/VideoView.tsx

```diff
@@ -21,17 +22,25 @@ import { GetVideo, GetVideoVariables } from '@/api/queries/__generated__/GetVide
 import { formatVideoViewsAndDate } from '@/utils/video'
 import { AddVideoView, AddVideoViewVariables } from '@/api/queries/__generated__/AddVideoView'
 import { ChannelLink } from '@/components'
+import { useInterval, usePersonalData } from '@/hooks'
 
 const VideoView: React.FC<RouteComponentProps> = () => {
   const { id } = useParams()
   const { loading, data, error } = useQuery<GetVideo, GetVideoVariables>(GET_VIDEO, {
     variables: { id },
   })
   const [addVideoView] = useMutation<AddVideoView, AddVideoViewVariables>(ADD_VIDEO_VIEW)
+  const { state, updateWatchedVideos } = usePersonalData()
 
+  const videoTimestamp = useMemo(() => {
+    const currentVideo = state.watchedVideos.find((v) => v.id === data?.video?.id)
+    return currentVideo?.__typename === 'INTERRUPTED' ? currentVideo.timestamp : 0
+  }, [data?.video?.id, state.watchedVideos])
+
+  const videoRef = useRef<HTMLVideoElement>(null)
   const videoID = data?.video?.id
 
-  const [playing, setPlaying] = useState<boolean>(true)
+  const [playing, setPlaying] = useState(false)
```

Comment:
This will basically remove the autoplay, so we need to do something about it. Since we're controlling the playback state outside of the player, we can remove the `autoplay` prop and something like this:

```tsx
const handlePlay = () => {
  setVideoPlaying(true)
}

const handlePause = () => {
  setVideoPlaying(false)
}

// ...

<VideoPlayer
  // ...
  playing={videoPlaying}
  onPlay={handlePlay}
  onPause={handlePause}
/>
```